crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -716,16 +716,31 @@ where
         {
             Some(parent_hash)
         } else {
-            // TODO: attempt to iterate over ancestors in the invalid cache
+            // iterate over ancestors in the invalid cache
             // until we encounter the first valid ancestor
+            let mut current_hash = parent_hash;
+            let mut current_header = self.invalid_headers.get(&current_hash);
+            while let Some(header) = current_header {
+                current_hash = header.parent_hash;
+                current_header = self.invalid_headers.get(&current_hash);
+
+                // If current_header is None, then the current_hash does not have an invalid
+                // ancestor in the cache, check its presence in blockchain tree
+                if current_header.is_none() {
+                    match self.blockchain.find_block_by_hash(current_hash, BlockSource::Any) {
+                        Ok(Some(_)) => return Some(current_hash),
```

Comment:
not a fan that we are ignoring a potentially critical error here (same as above), but we can address this separately. mb leave a todo?

Reply:
sure, done here https://github.com/paradigmxyz/reth/pull/8044/commits/6b5db27cfd0573f3bbe8bd5e7d80f502d258208b